docs(best_practices): add instruction how to redact sensitive data (#2265)
SumoLogic/sumologic-kubernetes-collection: deploy/docs/Best_Practices.md

              key: k8s_container_name
```

> ### Difference between /var/log/pods and /var/log/containers

Reviewer: Why this is subsection of "Overriding chart resource names with `fullnameOverride"?
It seems that you should remove one `#`

Reviewer: The title of this section is misleading. In this section it is described how `fluentd.tag` is  built and how it can be used, not the difference between /var/log/pods and /var/log/containers

Author: This is subsection of `## Collecting logs from /var/log/pods`
Not sure if this should go as separate section

Reviewer: Suggestion for the title: Form of `fluentd.tag` for /var/log/pods and /var/log/containers

Author: changed to `Fluentd tag for /var/log/pods and /var/log/containers`

> The only but important difference is generated `fluent.tag` value:

Reviewer 2: ```suggestion
The only, but important difference is the generated `fluent.tag` value:
```

Refer to our [documentation](v1_conf_examples.md) for other examples of how you can
customize the fluentd pipeline.

## Modifying logs in Fluentd

You can redact log messages in order to e.g. prevent sending sensitive data like passwords.

In order to do that [record_transformer filter][record_transformer] can be used.
Please consider the following configuration:

```yaml
fluentd:
  logs:
    containers:
      extraFilterPluginConf: |-

>           <filter containers.var.log.pods.*_*_*.test-container.*.log containers.var.log.containers.*_*_test-container-*.log>

Reviewer: In this section it is not mentioned that this filter is only for container with name `test-container`.

Author: added

            @type record_transformer
            enable_ruby
            <record>
              # Replace `Password: <pass>` with `Password: ***`
              log ${record["log"].gsub(/Password: (.*)/, 'Password: ***')}
              # Replace kubernetes['namespace_name'] with 'REDACTED'
              # if namespace_name exists in record['kubernetes'], then change value of it and then return modified record['kubernetes'] to assign as kubernetes
              kubernetes = ${if record['kubernetes'].has_key?('namespace_name'); record['kubernetes']['namespace_name'] = "REDACTED"; end; record['kubernetes']}
            </record>
          </filter>
```

It is going to replace all `Password: <pass>` occurence in logs with `Password: ***` and replace namespace to `REDACTED`

> for [all containers](#difference-between-varlogpods-and-varlogcontainers) named `multiline-logs-generator`.

Reviewer: Above configuration is for modifying logs from container with name `test-container`.

Author: updated

> for [all containers](#difference-between-varlogpods-and-varlogcontainers) named `multiline-logs-generator`.

Author: ```suggestion
for [all containers](#difference-between-varlogpods-and-varlogcontainers) named `test-container`.
```

An example log before entering the `extraFilterPluginConf` section is presented below:

```json
{
  "stream": "stdout",
  "logtag": "F",

>   "log": "log... ",

Reviewer: It will be good to have log with log message containing e.g "Password: 123456"

  "docker": {
    "container_id": "a1acfd70-c8d1-456b-95dd-515f1256906f"
  },
  "kubernetes": {
    "container_name": "example-container-multiline-logs-long-lines",

>     "namespace_name": "multiline-logs-generator",

Reviewer: I know that this is just example but it will better to have this example more related to above configuration so I'd prefer container name set to `test-container`.

Author: updated

Refer to our [documentation](v1_conf_examples.md) for other examples of how you can
customize the fluentd pipeline.

## Modifying logs in Fluentd

You can redact log messages in order to e.g. prevent sending sensitive data like passwords.

In order to do that [record_transformer filter][record_transformer] can be used.
Please consider the following configuration:

```yaml
fluentd:
  logs:
    containers:
      extraFilterPluginConf: |-

>           # Filter all test-container containers

Author: ```suggestion
          # Apply to all test-container containers
```
